packages/protocol/congestioncontrol/icca/scheduler/scheduler.go

```diff
@@ -556,7 +582,20 @@ func (s *Scheduler) updateDeficit(issuerID identity.ID, d *big.Rat) {
 
 	s.deficitsMutex.Lock()
 	defer s.deficitsMutex.Unlock()
-	s.deficits.Set(issuerID, minRat(deficit, MaxDeficit))
+	s.deficits.Set(issuerID, minRat(deficit, s.optsMaxDeficit))
+
+}
+
+func (s *Scheduler) GetExcessDeficit(issuerID identity.ID) (float64, error) {
+	var deficitFloat float64
+	s.deficitsMutex.RLock()
+	defer s.deficitsMutex.RUnlock()
+	if deficit, exists := s.deficits.Get(issuerID); exists {
+		deficitFloat, _ = deficit.Float64()
+	} else {
+		return 0.0, errors.New(fmt.Sprintf("Deficit for issuer %s does not exist", issuerID))
+	}
+	return deficitFloat - float64(s.IssuerQueueWork(issuerID)), nil
```

Comment:
I think the following is cleaner.

```suggestion
func (s *Scheduler) GetExcessDeficit(issuerID identity.ID) (deficitFloat float64, err error) {
	s.deficitsMutex.RLock()
	defer s.deficitsMutex.RUnlock()

	if deficit, exists := s.deficits.Get(issuerID); !exists {
		deficitFloat, _ = deficit.Float64()
		return deficitFloat - float64(s.IssuerQueueWork(issuerID)), nil
	}

	return 0.0, errors.New(fmt.Sprintf("Deficit for issuer %s does not exist", issuerID))
```